### PR TITLE
Adding store wakuv2 test

### LINKED ISSUE #4317: Add test for WakuV2 Store
There is a `wakuv2.Config.EnableStore` flag that can be used to enable Store protocol on local Status node:
https://github.com/status-im/status-go/blob/010afb4b39e0d4aa2315cc438c0c5759a5742f6d/wakuv2/waku.go#L306

However, there are no associated tests. It would be good to have one in a manner similar to `TestBasicWakuV2` here: https://github.com/status-im/status-go/blob/010afb4b39e0d4aa2315cc438c0c5759a5742f6d/wakuv2/waku_test.go#L176

but using a local store node.

### PR DESCRIPTION
**Pull request by plopezlpz**
Adding a store wakuv2 test
Closes #4317

### CODE REVIEW

File: wakuv2/waku_test.go [hunk 1]
```diff
@@ -280,3 +282,103 @@ func TestWakuV2Filter(t *testing.T) {
...
+
+func TestWakuV2Store(t *testing.T) {
+	// Configuration for the first Waku node
+	sql1, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	require.NoError(t, err)
+	config1 := &Config{
+		Port:              0,
+		EnableDiscV5:      false,
+		DiscoveryLimit:    20,
+		EnableStore:       true,
```
Comment (vitvly) [inline wakuv2/waku_test.go]: Do we need to enable store on the first Waku node?

File: wakuv2/waku_test.go [hunk 2]
```diff
@@ -280,3 +282,103 @@ func TestWakuV2Filter(t *testing.T) {
...
+	err = tt.RetryWithBackOff(func() error {
+		if len(w1.Peers()) == 0 {
+			return errors.New("no peers discovered")
+		}
+		return nil
+	}, options)
+	require.NoError(t, err)
+
+	// Wait for the nodes to discover each other
+	time.Sleep(1 * time.Second)
```
Comment (vitvly) [inline wakuv2/waku_test.go]: We use `Sleep` and `RetryWithBackOff` to wait for peers for discovered. I think this is often used in other tests too, however, just realised that we have this: https://github.com/status-im/status-go/blob/8e7d0a523e803f5f3b62f9b04deb3dd67449da16/wakuv2/waku.go#L1192
One can add a connection status subscription via `Waku.SubscribeToConnStatusChanges` and wait for `ConnStatus` structs to arrive.

File: wakuv2/waku_test.go [hunk 3]
```diff
@@ -280,3 +282,103 @@ func TestWakuV2Filter(t *testing.T) {
...
+	_, err = w1.Send(relay.DefaultWakuTopic, &pb.WakuMessage{
+		Payload:      []byte{1, 2, 3, 4, 5},
+		ContentTopic: contentTopic.ContentTopic(),
+		Version:      0,
+		Timestamp:    msgTimestamp,
+	})
+	require.NoError(t, err)
+
+	// Wait for the message to be transferred
+	time.Sleep(1 * time.Second)
```
Comment (vitvly) [inline wakuv2/waku_test.go]: In a similar fashion to connection status updates, we have this for envelope events: https://github.com/status-im/status-go/blob/8e7d0a523e803f5f3b62f9b04deb3dd67449da16/wakuv2/waku.go#L1419
One can subscribe to these envelope events via `Waku.SubscribeEnvelopeEvents`

Comment (vitvly) [reply wakuv2/waku_test.go]: So maybe this way we can obviate the need for `Sleep` statements.

**Review by vitvly — CHANGES_REQUESTED** [review]
Looks great, just some suggestions on possible ways to remove sleeps/retries in test code.

File: wakuv2/waku_test.go (hunk 2 above)
Comment (plopezlpz) [reply wakuv2/waku_test.go]: awesome, I'll try that

**Review by vitvly — APPROVED** [review]

**Pull request merged**